python/testmonitor/results/results.py

```diff
@@ -0,0 +1,224 @@
+
+This is an example of uploading test results to the SystemLink Test Monitor service.
+It simulates measuring the power output from a device and tests the measured power
+to ensure it is within a specified upper and lower limit.  The power is simulated using
+a the simple electrical equation P=VI (power=voltage*current).  In this example, a random
+amount of current loss and voltage loss are induced to simulate a non-ideal device.
+
+A top level result is created containing metadata about the overall test.
+
+The example sweeps across a range of input currents and voltages and takes measurements
+for each combination and stores a single measurement within each test step.  The test
+steps are associated with the test result, and in some cases, as child relationships
+to other test steps.  Each step is uploaded to the SystemLink server as it is generated.
+At the end, the step status is evaluated to set the status of the parent step and
+ultimately sets the status of the top-level test result.
+"""
+
+import uuid
+import random
+
+from systemlink.clients.nitestmonitor import ResultsApi
+from systemlink.clients.nitestmonitor import StepsApi
+from systemlink.clients.nitestmonitor.models.create_test_results_request import CreateTestResultsRequest
+from systemlink.clients.nitestmonitor.models.update_test_results_request import UpdateTestResultsRequest
+from systemlink.clients.nitestmonitor.models.test_step_create_or_update_request_object import \
+    TestStepCreateOrUpdateRequestObject
+from systemlink.clients.nitestmonitor.models.test_result_request_object import TestResultRequestObject
+from systemlink.clients.nitestmonitor.models.test_step_request_object import TestStepRequestObject
+from systemlink.clients.nitestmonitor.models.status_object import StatusObject
+from systemlink.clients.nitestmonitor.models.named_value_object import NamedValueObject
+
+
+def measure_power(current: float, voltage: float = 0) -> object:
+    """
+    Simulates taking an electrical power measurement.
+    This introduces some random current and voltage loss.
+    :param current: The electrical current value.
+    :param voltage: The electrical voltage value.
+    :return: A tuple containing the electrical power measurements and the input and output lists.
+    """
+    current_loss = 1 - random.uniform(0, 1) * 0.25
+    voltage_loss = 1 - random.uniform(0, 1) * 0.25
+    power = current * current_loss * voltage * voltage_loss
+
+    # Record electrical current and voltage as inputs.
+    inputs = [
+        NamedValueObject(name="current", value=current),
+        NamedValueObject(name="voltage", value=voltage)
+    ]
+
+    # Record electrical power as an output.
+    outputs = [
+        NamedValueObject(name="power", value=power)
+    ]
+
+    return power, inputs, outputs
+
+
+def build_power_measurement_params(
+        power: float,
+        low_limit: float,
+        high_limit: float,
+        status: object) -> object:
```

Comment:
We already imported `StatusObject`, so we could annotate this as `StatusObject`.

Reply:
I wasn't sure how to do type annotations for custom classes.  Is it simply replacing `"object"` with `"StatusObject"`?